### ISSUE TITLE
React wrapper?

### ISSUE THREAD

**Issue by sammyjava** [ZeLonewolf/openstreetmap-americana#442]
Not an issue per se, but I've got an Apollo GraphQL site running under React, now with OSM and Leaflet (which has the react-leaflet and react-leaflet-kml wrappers). I absolutely adore the americana style and would love if anyone could provide guidance on making it React-compatible. It's a lot of Javascript and I'm super weak on Javascript in general, having been strictly a server-side guy until this site rebuild.  Much appreciated!

https://pure-gas.org/maps/NM

This site gets a fair bit of dedicated traffic and the audience will scream with delight if they get proper highway markings!

**Comment by zekefarwell** [comment]
This project is not yet set up to allow the style to be easily used in other projects, although that is an eventual goal.  See #146.  Mostly likely we'd aim for a standard approach that would be usable on any type of site, regardless if using a specific frontend framework (React, Vue, Svelte, etc) or vanilla JS.  I'm sure many people [use MapLibre with React ](https://documentation.maptiler.com/hc/en-us/articles/4405444890897-How-to-display-MapLibre-GL-JS-map-using-React-JS)though.

**Comment by sammyjava** [comment]
Yeah I've got a MapLibre implementation using [react-map-gl](https://visgl.github.io/react-map-gl/) plus [maplibre-gl](https://visgl.github.io/react-map-gl/) which worked out pretty nice. The reason I went over to Leaflet is that my app places markers from KML files (which get updated every five minutes) and the most excellent [react-leaflet-kml](https://github.com/aviklai/react-leaflet-kml) is a thing that exists (and @aviklai updated it to work with current React/Leaflet versions in hours after I posted up, which always warms the heart!). So I got something working well (although I kind of wanted to use a vector base map and Leaflet is mainly raster support, but the download size doesn't differ much, and the OSM base layer is the way to freely go for a website that's about gas stations).

Are you saying I could be off and running americana with **react-map-gl**? I tried whaling on that for a while and gave up, admitting my extreme weakness with wrangling JS and opting for a beer instead.
